Test/BusinessLayer/Services/StudentsService.cs

```diff
@@ -0,0 +1,61 @@
+﻿using BusinessLayer.Models;
+using DataAccessLayer.Repositories;
+
+namespace BusinessLayer.Services
+{
+    public class StudentsService : IStudentsService{
+
+
+        private readonly IStudentsRepository studentsRepository;
+
+        public StudentsService(IStudentsRepository studentsRepository)
+        {
+            this.studentsRepository = studentsRepository;
+        }
+
+
+        //get all students
+        public async Task<IEnumerable<Student>> GetUsers()
+        {
+           return await studentsRepository.GetStudents();
+        }
+
+        //get student by id
+        public async Task<Student>  GetStudentByID(int id)
+        {
+            try
+            {
+                return await studentsRepository.GetStudentByID(id);
+            }
+            catch(Exception) 
+            {
+                throw;
```

Comment:
a specific error should be thrown.
https://www.loginradius.com/blog/engineering/exception-handling-in-csharp/

*to be applied everywhere it's needed in this file